src/toil_lib/pipelineWrapperBuilder.py

```diff
@@ -0,0 +1,160 @@
+
+class PipelineWrapperBuilder(object):
+    """
+    This class can be used to define wrapper scripts to run specific Toil pipelines in docker
+    containers. The purpose of this class is to provide a convenient way to define a command line
+    interface for a wrapper script and the logic to run the pipeline with this interface.
+    """
+    def __init__(self, name, desc, config, add_no_clean=True, add_resume=True):
+        """
+        :param str name: The name of the command to start the workflow.
+        :param str desc: The description of the workflow.
+        :param str config: A format string where each key exactly matches the name of an argument
+            defined in the arg_builder context manager. Note that dashes in argument names should be
+            changed to underscores in this string e.g. 'no-clean' should be 'no_clean'.
+        """
+        self._name = name
+        self._desc = desc
+        self._config = config
+        self._no_clean = add_no_clean
+        self._resume = add_resume
+
+    def run(self, args, pipeline_command):
+        """
+        Invokes the pipeline with the defined command. Command line arguments, and the command need
+        to be set with arg_builder, and command_builder respectively before this method can be
+        invoked.
+        """
+        # prepare workdir
+        mount = self._prepare_mount(args)
+        self._workdir = os.path.join(mount, 'Toil-' + self._name)
+        # prepare config
+        args_dict = vars(args)
+        args_dict['output_dir'] = mount
+        self._config = textwrap.dedent(self._config.format(**args_dict))
+        config_path = os.path.join(self._workdir, 'config')
+        command = self._make_command_prefix(os.path.join(self._workdir, 'jobStore'),
+                                            config_path,
+                                            self._workdir) + pipeline_command
+        if self._resume and args.resume:
+            command.append('--restart')
+        self._create_workdir(args)
+        with open(config_path, 'w') as f:
+            f.write(self._config)
+
+        try:
+            subprocess.check_call(command)
+        except subprocess.CalledProcessError as e:
+            print(e, file=sys.stderr)
+        finally:
+            log.info('Pipeline terminated, changing ownership of output files from root to user.')
+            stat = os.stat(self._mount)
+            subprocess.check_call(['chown', '-R', '{}:{}'.format(stat.st_uid, stat.st_gid),
+                                   self._mount])
+            if self._no_clean and args.no_clean:
+                log.info('Flag "--no-clean" was used, therefore %s was not deleted.', self._workdir)
+            else:
+                log.info('Cleaning up temporary directory: %s', self._workdir)
+                shutil.rmtree(self._workdir)
+
+    def get_args(self):
+        """
+        Use this context manager to add arguments to an argparse object with the add_argument
```

Comment:
Is this still a context manager?  I thought a contextmanager needed a `@contextmanager` decorator or needed to be a class with `__enter__` and  `__exit__` method. 

Can you provide an example of how someone would add an argument via this method?